src/Neo/SmartContract/Native/NativeContract.cs

```diff
@@ -266,19 +266,21 @@ public ContractState GetContractState(IsHardforkEnabledDelegate hfChecker, uint
         /// </summary>
         /// <param name="settings">The <see cref="ProtocolSettings"/> where the HardForks are configured.</param>
         /// <param name="index">Block index</param>
-        /// <param name="hardfork">Active hardfork</param>
+        /// <param name="hardforks">Active hardforks</param>
         /// <returns>True if the native contract must be initialized</returns>
-        internal bool IsInitializeBlock(ProtocolSettings settings, uint index, out Hardfork? hardfork)
+        internal bool IsInitializeBlock(ProtocolSettings settings, uint index, out Hardfork[]? hardforks)
```

Comment:
```suggestion
        internal bool TryInitializeBlock(ProtocolSettings settings, uint index, out Hardfork[] hardforks)
```

Please dont add this `?` `nullable` until `nullable` is enabled on in `*.csproj`.